sphinx/parsers.py

```diff
@@ -20,6 +19,32 @@
     from sphinx.application import Sphinx
 
 
+class RSTStateMachine(docutils.parsers.rst.states.RSTStateMachine):
+    def __init__(self, app: Sphinx, state_classes: Sequence[type[State]], initial_state: str,
+                 debug: bool = False):
+        self.app = app
+        super().__init__(state_classes=state_classes, initial_state=initial_state,
+                         debug=debug)
+
+    def insert_input(self, include_lines, path):
+        # First we need to combine the lines back into text so we can send it with the
+        # source-read event. In newer releases of docutils there are two lines at the end,
```

Comment:
The change to affix the marker in docutils was fortunately already there in version 0.18, as you can see [here](https://sourceforge.net/p/docutils/code/HEAD/tree/tags/docutils-0.18/docutils/parsers/rst/directives/misc.py#l196).  So there won't be a compatibility issue with the version range you listed.

Reply:
Good then. If you can point which docutils version introduced this feature, maybe you could update the comment by saying "In docutils x.y+ [...]" (at least, future maintainers wouldn't ask themselves "which docutils version?"). If it's too much work, let's put something like "at least since 0.18".

Reply:
Yes, no problem.  The change to affix the include marker was introduced in 0.18.  I will update the comment with that information.